core/controllers/moderator.py

```diff
@@ -78,15 +78,60 @@ def post(self) -> None:
             'featured_activity_reference_dicts']
 
         try:
-            summary_services.require_activities_to_be_public(
+            # Retrieve the list for each type of inavlid ID.
+            dne_exp, dne_col, priv_exp, priv_col = (
+                summary_services.check_activity_id_validity(
                 featured_activity_references)
-        except Exception as e:
-            raise self.InvalidInputException(e)
+            )
 
-        activity_services.update_featured_activity_references(
-            featured_activity_references)
+            # If all of the lists are empty, there are no invalid IDs.
+            if ((not dne_exp) & (not dne_col) &
+                (not priv_exp) & (not priv_col)):
+                activity_services.update_featured_activity_references(
+                featured_activity_references)
+                self.render_json({})
+            else:
+
+                error_message = ''
+
+                # If there are IDs for non-existent Explorations.
+                if dne_exp:
+                    error = (
+                        f'These Exploration IDs do not exist: '
+                        f'{", ".join(str(id) for id in dne_exp)}. '
+                    )
+                    error_message = error_message + error
+
+                # If there are IDs for non-existent Collections.
+                if dne_col:
+                    error = (
+                        f'These Collection IDs do not exist: '
+                        f'{", ".join(str(id) for id in dne_col)}. '
+                    )
+                    error_message = error_message + error
+
+                # If there are IDs for private Explorations.
```

Comment:
Here and many of the code comments are not required. If you still think this can improve understanding, please provide more context for the code comments.

_(I find little hard to understand some of the comments below )_